sous-chef/sous_chef/date/get_due_date.py

```diff
@@ -77,14 +78,27 @@ def replace_time_with_meal_time(
         return self._set_specified_time(due_date, hour, minute)
 
     @staticmethod
-    def _get_anchor_date_at_midnight(weekday: str) -> datetime.datetime:
+    def _get_anchor_date_at_midnight(
+        weekday: str, week_offset: int = 1
+    ) -> datetime.datetime:
         weekday_index = get_weekday_index(weekday)
         today = datetime.date.today()
         today_index = today.weekday()
-        if today_index >= weekday_index:
+
+        # by default (week_offset=1) anchor_day will be in the upcoming week,
+        # independent of the current weekday.
+        # week_offset can be set to an arbitrarily higher (or smaller) number.
+        # an anchor day in the same week can be enforced with week_offset=0
+        # unless that day has already passed,
+        # then it will switch to the next week
+        # = same behaviour as with week_offset=1
+        # if a past anchor day is really desired (not an expected use case),
+        # a negative value of week_offset can be specified
+        if (today_index > weekday_index) and week_offset < 1:
```

Comment:
Why not using self.week_offset here?

Reply:
This is in a static method.

Reply:
change to non-static method please & fix test if needed